include/tvm/tir/builtin.h

```diff
@@ -720,6 +720,16 @@ TVM_DLL const Op& texture2d_load();
  */
 TVM_DLL const Op& mem_copy();
 
+/*!
+ * \brief Initiate a non-blocking DMA copy from source to destination
+ */
+TVM_DLL const Op& dma_copy();
+
+/*!
+ * \brief Wait until the number of DMAs in flight is less than or equal to some maximum
+ */
+TVM_DLL const Op& dma_wait();
```

Comment:
I think it is fine to have Hexagon-specific intrinsics here, rather than trying to be more generic than necessary.

For CUDA we have specific copy / wait intrins in https://github.com/apache/tvm/blob/c4aab62c6d588ed06f60b0db9691271f5883dc66/include/tvm/tir/builtin.h#L635-L652

Reply:
Right, I have attempted to make the pass / intrinsics generic.  Perhaps that is not correct.  We could leave as-is given CI is passing.  Or, we could do _one or both_ of the following:

1) Make the pass / intrinsics specific to Hexagon
2) Add user level control to opt in to running this pass

Reply:
I am open to making it a Hexagon only pass, but not quite sure how to do it.  Investigating.

Reply:
For CUDA async copy, I added an option that users need to enable manually:

https://github.com/apache/tvm/blob/036aa722aefd7126c17063cddb260b9347f17598/src/driver/driver_api.cc#L546-L551

For Hexagon, we may just enable this pass in `driver_api.cc` whenever the target is Hexagon.

Reply:
@masahi what do you think about generalizing that switch by calling it `use_async_copy` and having both the CUDA pass and my new `LowerAsyncDMA` pass use the same switch?

Reply:
Yeah I like that. And for Hexagon we can enable that switch by default.